### PR TITLE
Code cleanup! [FGTS-75] [FGTS-94]

### LINKED ISSUE #12: Change all Player behaviour timers into actual Timer nodes
At the moment, much of the player's behaviour is controlled by timers declared as floating point numbers in the code, which are manually updated each tick. We've decided to change these into actual Timer nodes in the Player scene.

Notes about timers:
- You may not need to use the signal emitted by the timer if you're only trying to check whether the timer has completed or not. In that case I believe it's better to just grab the time_left off of the Timer node and check whether there is time left or if it has completed.
- If your timer is controlling game element logic (ex. player or enemy logic) then please set the timer's Process Mode to Physics instead of the default of idle. This should make it behave more consistently. Only use Process Mode = Idle for animations or sound effects (nothing that could affect anything physics-related).
- If you will always set the timer running by specifying the duration, then don't bother setting it in the scene. We should probably agree to always do this and to define the timer duration as a variable at the top of the file.

Starting timers: `jump_frame_countdown = jump_length` => `$JumpCountdownTimer.start(jump_length)`
Stopping timers: `jump_frame_countdown = 0` => `$JumpCountdownTimer.stop()`
Checking remaining time on the timer: `jump_frame_countdown > 0` => `$JumpCountdownTimer.time_left > 0`
Checking if timer is done or not: $JumpCountdownTimer.is_stopped() : returns a boolean

### PR DESCRIPTION
**Pull request by charasyn**
- Fix warnings with unused function parameters throughout code
- Switch player code to use timer objects instead of float variables
- Remove debug print from player swing
- Fix error message printed during gameplay [FGTS-94]
- Fix error message during swinging [FGTS-75]

### CODE REVIEW

File: godotproject/Player.gd [hunk 1]
```diff
@@ -15,9 +15,9 @@ export (float) var air_turn_coefficient = 1.2
 
 export (float) var jump_leniency_time = 5 / 60.0
 export (float) var jump_speed = 500
-export (float) var jump_length = 10 / 60.0
+export (float) var jump_length = (10+1) / 60.0
 export (float) var shorthop_speed = jump_speed * 0.6
-export (float) var shorthop_length = 6 / 60.0
+export (float) var shorthop_length = (6+1) / 60.0
```
Comment (charasyn) [inline godotproject/Player.gd:20]: I think the order between checking and adjusting was changed now that we're using a timer:
Old:
```
if jump timer is not done:
  decrease timer
  set player Y velocity to jump speed
```
New:
```
decrease timer (done by timer object)
...
if jump timer is not done:
  set player Y velocity
```
In order to keep the jump height the same, I had to add an extra frame to the jump duration.

File: godotproject/Player.gd [hunk 2]
```diff
@@ -45,10 +45,6 @@ onready var animation_tree = get_node("AnimationTree")
 onready var animation_mode = animation_tree.get("parameters/playback")
 
 var velocity = Vector2.ZERO
-var jump_frame_countdown = 0
-var jump_shorthop_countdown = 0
-var jump_button_press_leniency_timer = 0
-var jump_on_ground_leniency_timer = 0
```
Comment (charasyn) [inline godotproject/Player.gd:51]: These were all converted to timer objects :)

File: godotproject/Player.gd [hunk 3]
```diff
@@ -229,7 +219,8 @@ func _draw():
 	elif $PlayerTongue.swinging:
 		tongue_target_global = swing_pivot_position
 	
-	$Line2D.remove_point(1)
+	if $Line2D.get_point_count() > 1:
+		$Line2D.remove_point(1)
```
Comment (charasyn) [inline godotproject/Player.gd]: This fixes the error that is printed every frame - we only remove the point if it exists. [FGTS-94]

**Review by charasyn — COMMENTED** [review]
A few comments I wanted to put to provide more context.

**Review by fheysel — APPROVED** [review]

**Comment by charasyn** [comment]
Closes #12

**Pull request merged**